### PR TITLE
Default file extension for Open file dialog should be .py

### LINKED ISSUE #3436: Default file extension should be .py instead of empty
## The default file type is empty in the save dialogs

**What steps will reproduce the problem?**

1. Open any document and use save as dialog
2. if there exists any remove the extension
3. Click save

**What is the expected output? What do you see instead?**

The expected outcome (I think) the file should be save with a defauly .py extension instead of no-extension state. 

**Please provide any additional information below**

You can see from the screenshot possible extension recommendation windows is also empty

zzzz

## Versions and main components

* Spyder Version: 3.0.0b7
* Python Version: 3.6
* Operating system: Win10

## Dependencies
jedi >=0.8.1     :  0.9.0 (OK)
matplotlib >=1.0 :  1.5.1 (OK)
nbconvert >=4.0  :  4.2.0 (OK)
numpy >=1.7      :  1.11.1 (OK)
pandas >=0.13.1  :  0.18.1 (OK)
pep8 >=0.6       :  1.7.0 (OK)
pyflakes >=0.6.0 :  1.3.0 (OK)
pygments >=2.0   :  2.1.3 (OK)
pylint >=0.25    :  1.6.4 (OK)
qtconsole >=4.2.0:  4.2.1 (OK)
rope >=0.9.4     :  0.9.4-1 (OK)
sphinx >=0.6.6   :  1.4.6 (OK)
sympy >=0.7.3    :  None (NOK)

### PR DESCRIPTION
**Pull request by malliwi88**
Fixes #3436

### CODE REVIEW

File: spyder/widgets/editor.py [hunk 1]
```diff
@@ -25,13 +25,15 @@
 from qtpy.QtGui import QFont, QKeySequence
 from qtpy.QtWidgets import (QAction, QApplication, QHBoxLayout, QMainWindow,
                             QMessageBox, QMenu, QSplitter, QVBoxLayout,
-                            QWidget)
+                            QWidget, QFileDialog)
```
Comment (goanpeca) [inline spyder/widgets/editor.py]: Why did you add QFileDialog? it seems to not be used

Comment (goanpeca) [inline spyder/widgets/editor.py]: Oops my bad, I see where you used it

File: spyder/widgets/editor.py [hunk 2]
```diff
@@ -1309,11 +1311,24 @@ def select_savename(self, original_filename):
...
             # Save the currently edited file
+            if not self.get_stack_count():
+                return
             index = self.get_stack_index()
+
         finfo = self.data[index]
-        filename = self.select_savename(finfo.filename)
+
+        self.redirect_stdio.emit(False)
+        selectedfilter = ''
```
Comment (goanpeca) [inline spyder/widgets/editor.py]: Why is selected filter an empty string?

File: spyder/plugins/editor.py [hunk 3]
```diff
@@ -1887,14 +1888,9 @@ def save(self, index=None, force=False):
...
     
     @Slot()
-    def save_as(self):
-        """Save *as* the currently edited file"""
-        editorstack = self.get_current_editorstack()
-        if editorstack.save_as():
-            fname = editorstack.get_current_filename()
-            if CONF.get('workingdir', 'editor/save/auto_set_to_basedir'):
-                self.open_dir.emit(osp.dirname(fname))
-            self.__add_recent_file(fname)
```
Comment (goanpeca) [inline spyder/plugins/editor.py]: Why did you remove all this logic?

File: spyder/widgets/editor.py [hunk 4]
```diff
@@ -1325,10 +1340,6 @@ def save_as(self, index=None):
 
             new_index = self.rename_in_data(index, new_filename=filename)
 
-            # We pass self object ID as a QString, because otherwise it would
-            # depend on the platform: long for 64bit, int for 32bit. Replacing
-            # by long all the time is not working on some 32bit platforms
-            # (see Issue 1094, Issue 1098)
```
Comment (goanpeca) [inline spyder/widgets/editor.py]: Why did you remove that comment explaining why the changes are needed ?

Comment (malliwi88) [inline spyder/widgets/editor.py]: Hi @goanpeca these comments do not belong to my amendment, so I preferred to remove trash.

Comment (goanpeca) [inline spyder/widgets/editor.py]: >Hi @goanpeca these comments do not belong to my amendment, so I preferred to remove trash.

If you did not made those comments, and the comments do not refer to code you are modifying, how is that trash??? please undo those changes. You should never remove comments of work that is not related to your work

File: spyder/plugins/editor.py (hunk 3 above)
Comment (malliwi88) [inline spyder/plugins/editor.py]: @goanpeca I remove it because integrate all logic on Widget module, And only call save_as functionality from Widget module with this sentence... on line 1893 (plugins/editor.py)
> `self.get_current_editorstack().save_as(index=index)`
